src/transformation/visitors/return.ts

```diff
@@ -6,13 +6,42 @@ import { validateAssignment } from "../utils/assignment-validation";
 import { createUnpackCall, wrapInTable } from "../utils/lua-ast";
 import { ScopeType, walkScopesUp } from "../utils/scope";
 import { isArrayType } from "../utils/typescript";
-import { isMultiFunction, transformMultiReturnStatement } from "./language-extensions/multi";
+import { transformArguments } from "./call";
+import {
+    returnsMultiType,
+    multiReturnCallShouldBeWrapped,
+    isMultiFunction,
+    isMultiReturnType,
+} from "./language-extensions/multi";
+import { invalidMultiFunctionReturnType } from "../utils/diagnostics";
 
 function transformExpressionsInReturn(
     context: TransformationContext,
     node: ts.Expression,
     insideTryCatch: boolean
 ): lua.Expression[] {
+    if (ts.isCallExpression(node)) {
+        // $multi(...)
+        if (isMultiFunction(context, node)) {
```

Comment:
This naming is a little confusing to me actually, you are checking if the function call isMultiFunction, I would expect node.expression (the name of the function itself) to be the multiFunction here though.

Reply:
I agree (this is how it was named already). I think `isMultiFunctionCall` would make more sense.